gradio/routes.py

```diff
@@ -222,15 +223,17 @@ def login(form_data: OAuth2PasswordRequestForm = Depends()):
         def main(request: fastapi.Request, user: str = Depends(get_current_user)):
             mimetypes.add_type("application/javascript", ".js")
             blocks = app.get_blocks()
+            root_path = request.scope.get("root_path")
```

Comment:
@abidlabs Thanks for the fix! If we're reading the root_path from the request, why do we need an argument in blocks?

Reply:
Good question. The `root_path` from the `request` will include the subpath if you mount Gradio within a FastAPI app. However, it will *not* include any subpath that comes from the nginx reverse proxy since the FastAPI app does not know about that. So you need to put both

Reply:
where is the Blocks.root_path used? I only see the root_path from the request being used.

Reply:
In `configure_app`. The FastAPI's app's `root_path` is set to  Blocks's `root_path`

Reply:
When you do `request.scope.get("root_path")`, it combines both the FastAPI app's explicit root path (the same one coming from the nginx reverse proxy), along with any subpath that your app may have because it's mounted within a larger FastAPI

Reply:
Take a look at this user's 2 examples to see why you need both: https://github.com/gradio-app/gradio/issues/3472#issuecomment-1541446133

Reply:
I see it being set to `self.root_path` but never being used, is that something the FastAPI app automatically recognizes?

Reply:
Yup the `root_path` is used internally by `request.scope.get("root_path")`. This is kinda mentioned here: https://fastapi.tiangolo.com/advanced/behind-a-proxy/#setting-the-root_path-in-the-fastapi-app